### DISCUSSION TITLE
Get an email notification when a new comment is added

### DISCUSSION THREAD

**Discussion by kimlundjohansen** [PiranhaCMS/piranha.core#2016] [Ideas] [answered]
Currently my setup is to not auto approve comments. To avoid having to closely monitor the admin interface for incoming awaiting approval it would be nice to retrieve an email for the registered SysAdmin.

**Comment by tidyui** [comment, marked as answer]
Hi there! The correct way to approach this would be to use the existing `OnAfterSave` hook for comments, see the docs here:

https://piranhacms.org/docs/master/application/hooks

This way you can add application code to send an email after every new comment is added.

Best regards

**Reply by kimlundjohansen** [reply]
Thanks, that seems like a fine approach.

**Reply by kimlundjohansen** [reply]
To have access to the Post data I ended up extending the SinglePostWithComments and overriding OnPostSaveComment. I would prefer the hook approach though, is it possible to access the Post data from App.Hooks.Comments.RegisterOnAfterSave ?
